src/models/step_arguments/data_table.js

```diff
@@ -30,4 +55,38 @@ export default class DataTable {
     }
     return _.fromPairs(rows)
   }
-}
+
+  typedRowsHash() {
+    const erroredRowFormat = ( () => {
+      const pad = '             '.split('').join('                     ')
+      const colPadder = () => {
+        let colLen = 0
+        return (v) => {
+          colLen = Math.max(colLen, v.length)
+          return {toString: () => (v + pad).substr(0, colLen)}
+        }
+      }
+      const cols = [colPadder(), colPadder(), colPadder()]
+      return (row) => {
+        return cols.map( (padder, i) => padder( row[i] ) )
+      }
+    })()
```

Comment:
the proposed data-structure is introducing complexity instead of keeping things flat and simple.
loops, really? (even if you don't write it - it still happens...)

How bad is it needed for you?